### ISSUE TITLE
Method Syndata+IC13+IC17

### ISSUE THREAD

**Issue by cong235** [backtime92/CRAFT-Reimplementation#2]
Can you share your .pth for this method ?

**Comment by backtime92** [comment]
@cong235 ok，I will upload the link,just wait for a moment

**Comment by cong235** [comment]
Hi @backtime92, your method just work on character level right?

**Comment by backtime92** [comment]
@cong235 No, we provides  the weak supervised traing methods and supports  for the word level based on the water labeling algorithm.

**Comment by cong235** [comment]
@backtime92 i got it. Thank you. i'll try to train and give you result. Hope it can help you

**Comment by backtime92** [comment]
@cong235 ok,I will release the dataloader code just a few days,thanks for your attention.

**Comment by cong235** [comment]
@backtime92 i got it. I will wait you here. Thank you

**Comment by cong235** [comment]
@backtime92 i was tested your water labeling algorithm. I see the result output it maybe not good because can split each word. in the original paper, they use weakly supervised learning for split word. So maybe it is reason we dont have good Precision like original paper.

**Comment by backtime92** [comment]
@cong235 You are right, I am try to adjust the hyper-parameters in water labeling algorithm and make it better to split the character. But I am not clear what you said about weakly supervised learning? In the author's paper, the weakly supervised learning is water labeling algorithm using to split the character. .

**Comment by cong235** [comment]
@backtime92 They mention it in part 3.2.2. Fig4. I'm so confuse when the first they said:"Here, we generate character boxes from each word-level annotation in a weaklysupervised manner, as summarized in Fig. 4" and later they said: "Fig. 6 shows the entire procedure for splitting the characters." and in Fig6 they used watershed algorithm for split box. Can you explain it ?

**Comment by backtime92** [comment]
Yes, because the icdar13-17 did not include the character level annotation, so the author used the model trained with syndata to generate the score map for the icdar image. Then water labeling split the character to get the box coordinates which can be generate gaussian map.

**Comment by cong235** [comment]
@backtime92 it mean, the syndata have character annotation ?

**Comment by cong235** [comment]
@backtime92 i was read the paper 1 more time and i see their flow for split box is : 
the first they will crop word from original image, 2nd they use trained model for predict region score and the third they use water for creat bbox for each region score. so i have 1 question, you was use thier pretrain model for predict region score before you use water right ? and another question, when you train your model you train from scratch or used pretrain ?

**Comment by backtime92** [comment]
@cong235 The pre-trained model was trained on Syndata which is Synthetic data

**Comment by cong235** [comment]
@backtime92 can i get it on internet ?

**Comment by backtime92** [comment]
@cong235 Yes, i have released the pre-trained model  link in Syndata and you can download it in baidu or google drive.

**Comment by cong235** [comment]
@backtime92 but can i get the dataset Syndata you was use?

**Comment by backtime92** [comment]
@cong235 Em.... You can download it in the Synthetich data paper whic have given the link.

**Comment by cong235** [comment]
@backtime92 thank you bro!

**Comment by backtime92** [comment]
@cong235 
1. No, the pre-trained model which I used is from torch.
2.No, as you can see, the for loop is to read IC15 data and the iter-next is to read Syndata. And you can see that I have used the torch.cat concatenate the Syndata and IC15 data.

**Comment by cong235** [comment]
@backtime92 Thank you. It mean you train with both data IC15 and Syndata.

**Comment by backtime92** [comment]
@cong235 yes

**Comment by cong235** [comment]
@backtime92 did you try with this dataset ? Because in other dataset just focus on latin-char so sometime it too bad for Chinese or Japanese
https://ctwdataset.github.io/

**Comment by cong235** [comment]
@backtime92 did you updated new method in currently code? because i cant see any change in your git.
"Note:new gaussian map method can split the inference gaussian region score map"

**Comment by backtime92** [comment]
@cong235 No, I am working on it.

**Comment by cong235** [comment]
@backtime92 I see in the method synth+ic15 i need put the data path for synthtext, but if i just want train with only ic15 because dont have data synthtext, can i do that?

**Comment by backtime92** [comment]
@cong235 Yes, just take out the syndata. And I think you can get can the syndata link, using the thunder to download.

**Comment by cong235** [comment]
@backtime92 it mean the syndata will use for training together with ic15 right ? because i can see the concat 2 data syn and ic15 before feed to model.

**Comment by backtime92** [comment]
@cong235 for index, (images, gh_label, gah_label, mask, _) in enumerate(real_data_loader):
            #real_images, real_gh_label, real_gah_label, real_mask = next(batch_real)
            #syn_images, syn_gh_label, syn_gah_label, syn_mask, __ = next(batch_syn)
            #images = torch.cat((syn_images,real_images), 0)
            #gh_label = torch.cat((syn_gh_label, real_gh_label), 0)
            #gah_label = torch.cat((syn_gah_label, real_gah_label), 0)
            #mask = torch.cat((syn_mask, real_mask), 0)

**Comment by cong235** [comment]
@backtime92 can i start training from your checkpoint with file "trainic15data.py" ?Because when i read your code it just load pretrain synth and start training with new checkpoint without keep training from old checkpoint.

**Comment by raghavgurbaxani** [comment]
@backtime92 @cong235 how can I get charachter level bounding boxes , I tried setting the --link_threshold to 99999 and replacing
text_score_comb = np.clip(text_score + link_score, 0, 1)
With this line

text_score_comb = np.clip(text_score, 0, 1)
as mentioned https://github.com/clovaai/CRAFT-pytorch/issues/46 

but I dont get charachter level boxes . PS I am using this model - https://drive.google.com/file/d/1SkJEfaGYIq-eFxfzFVZb-cGdGWR8lPSi/view

Thanks
